pennylane/hf/convert.py

```diff
@@ -0,0 +1,318 @@
+    0.2 [Y0 Z2]
+    """
+    all_wires = Wires.all_wires([op.wires for op in ops], sort=True)
+
+    if wires is not None:
+        qubit_indexed_wires = _process_wires(
+            wires,
+        )
+        if not set(all_wires).issubset(set(qubit_indexed_wires)):
+            raise ValueError("Supplied `wires` does not cover all wires defined in `ops`.")
+    else:
+        qubit_indexed_wires = all_wires
+
+    q_op = openfermion.QubitOperator()
+    for coeff, op in zip(coeffs, ops):
+
+        if not isinstance(op, qml.operation.Tensor):
+            op = qml.operation.Tensor(op)
+
+        extra_obsvbs = set(op.name) - {"PauliX", "PauliY", "PauliZ", "Identity"}
+        if extra_obsvbs != set():
+            raise ValueError(
+                f"Expected only PennyLane observables PauliX/Y/Z or Identity, "
+                f"but also got {extra_obsvbs}."
+            )
+
+        # Pauli axis names, note s[-1] expects only 'Pauli{X,Y,Z}'
+        pauli_names = [s[-1] if s != "Identity" else s for s in op.name]
+
+        all_identity = all(obs.name == "Identity" for obs in op.obs)
+        if (op.name == ["Identity"] and len(op.wires) == 1) or all_identity:
+            term_str = ""
+        else:
+            term_str = " ".join(
+                [
+                    f"{pauli}{qubit_indexed_wires.index(wire)}"
+                    for pauli, wire in zip(pauli_names, op.wires)
+                    if pauli != "Identity"
+                ]
+            )
+
+        # This is how one makes QubitOperator in OpenFermion
+        q_op += coeff * openfermion.QubitOperator(term_str)
+
+    return q_op
+
+
+def _openfermion_pennylane_equivalent(
+    openfermion_qubit_operator, pennylane_qubit_operator, wires=None
+):
+    r"""Checks equivalence between OpenFermion :class:`~.QubitOperator` and Pennylane  VQE
+    ``Hamiltonian`` (Tensor product of Pauli matrices).
+
+    Equality is based on OpenFermion :class:`~.QubitOperator`'s equality.
+
+    Args:
+        openfermion_qubit_operator (QubitOperator): OpenFermion qubit operator represented as
+            a Pauli summation
+        pennylane_qubit_operator (pennylane.Hamiltonian): PennyLane
+            Hamiltonian object
+        wires (Wires, list, tuple, dict): Custom wire mapping used to convert to qubit operator
+            from an observable terms measurable in a PennyLane ansatz.
+            For types Wires/list/tuple, each item in the iterable represents a wire label
+            corresponding to the qubit number equal to its index.
+            For type dict, only int-keyed dict (for qubit-to-wire conversion) is accepted.
+            If None, will map sorted wires from all Pennylane terms to consecutive int.
+
+    Returns:
+        (bool): True if equivalent
+    """
+    coeffs, ops = pennylane_qubit_operator.terms()
+    return openfermion_qubit_operator == _pennylane_to_openfermion(coeffs, ops, wires=wires)
+
+
+def import_observable(qubit_observable, wires=None, tol=1e08):
```

Comment:
perhaps this should be `import_operator`?

Reply:
Also the tolerance looks odd being so high, but I see it is in terms of machine epsilon 🤔

Reply:
Is the tolerance missing a `-` sign?

Reply:
`tol` indeed [seems to be](https://numpy.org/doc/stable/reference/generated/numpy.real_if_close.html) in the unit of machine epsilon. For `float32` this is about 1e-7, and about 1e-16 for `float64`. Not sure about the default choice here though.

Reply:
The `+` sign is correct and the defaults is apparently assumed to be `float64`. So the tolerance for `np.real_if_close` becomes `~1e-8`.

Reply:
I changed the function name from `import_observable` to `import_operator`.